share/wake/lib/system/remote_cache_api.wake

```diff
@@ -331,16 +333,53 @@ export def rscApiFindMatchingJob (req: CacheSearchRequest) (api: RemoteCacheApi)
 #  rscApiGetStringBlob (RemoteCacheBlob "asdf" "https://...") = Pass "foo\nbar\nbat" 
 # ```
 export def rscApiGetStringBlob ((CacheSearchBlob _ uri): CacheSearchBlob): Result String Error =
-    # TODO: use "small blob" for this and also parse the schema out
-    # require False = id ==* "00000000-0000-0000-0000-000000000000"
-    # else Pass  ""
+    require scheme, path, Nil = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
+    else failWithError "rsc: uri has too many components: '{uri}'"
```

Comment:
That does mean that if the URI doesn't specifically include a `://` substring we'll hit this error with a `Nil` returned, which is kind of the opposite of "too many components".